host/xtest/regression_1000.c

```diff
@@ -324,7 +327,141 @@ static void xtest_tee_test_1002(ADBG_Case_t *c)
 	TEEC_CloseSession(&session);
 }
 
+struct test_1003_arg {
+	uint32_t test_type;
+	size_t repeat;
+	size_t max_before_lockers;
+	size_t max_during_lockers;
+	size_t before_lockers;
+	size_t during_lockers;
+	TEEC_Result res;
+	uint32_t error_orig;
+};
+
+static void *test_1003_thread(void *arg)
+{
+	struct test_1003_arg *a = arg;
+	TEEC_Session session = { 0 };
+	size_t rounds = 64 * 1024;
```

Comment:
FYI, this is much too small to always reach the theoretical maximum number of read concurrency and read waiters (4) on HiKey960. But it doesn't really matter I suppose.

Reply:
Agree